src/reqwesp/client.rs

```diff
@@ -0,0 +1,70 @@
+use anyhow::Result;
+
+use embedded_svc::http::client::Client as HttpClient;
+use embedded_svc::http::Method;
+use esp_idf_svc::http::client::{Configuration as HttpConfig, EspHttpConnection};
+
+use crate::{Request, RequestBuilder, Response};
+
+pub struct Client {
+    client: HttpClient<EspHttpConnection>,
+}
+
+impl<'a> Client {
```

Comment:
Are we implementing an interface? I don't understand this syntax. Do we specify the class attributes in the `pub struct` and then implement the method in `Client`?

Reply:
Yes, you are correct. We are implementing the associated functions and methods for the `Client` struct.
In this case, `client` is the only attribute. You can see that this struct is a wrapper for the ESP HTTP client (`client`) to provide more flexible usage.

The `pub` keyword means it is public. All structs, fields, traits, etc. are private by default. If it was not public, then `main.rs` and `exampes/http.rs` would not be able to see it. You can also restrict the [visibility](https://doc.rust-lang.org/rust-by-example/mod/visibility.html) to only the files within the module if you use `pub(crate)`. For example, if I used `pub(crate)` on `Client` then `Client` can still be imported in `request.rs`, but cannot be imported in `main.rs`.